ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -565,49 +577,202 @@ toPublicState PeerSelectionState { knownPeers
+                                 -> Int -> Int -> Int -- bootstrap peers
+                                 -> Int -> Int -> Int -- local roots
+                                 -> [(HotValency, WarmValency)]
+                                 -> Int -> Int -> Int -- shared peers
+                                 -> PeerSelectionCounters
+pattern PeerSelectionCountersHWC { numberOfColdPeers,
+                                   numberOfWarmPeers,
+                                   numberOfHotPeers,
+
+                                   numberOfColdBigLedgerPeers,
+                                   numberOfWarmBigLedgerPeers,
+                                   numberOfHotBigLedgerPeers,
+
+                                   numberOfColdBootstrapPeers,
+                                   numberOfWarmBootstrapPeers,
+                                   numberOfHotBootstrapPeers,
+
+                                   numberOfColdLocalRootPeers,
+                                   numberOfWarmLocalRootPeers,
+                                   numberOfHotLocalRootPeers,
+                                   localRootsHWC,
+
+                                   numberOfColdSharedPeers,
+                                   numberOfWarmSharedPeers,
+                                   numberOfHotSharedPeers }
+        <- (peerSelectionCountersHWC ->
+             PeerSelectionCounters { numberOfKnownPeers                = numberOfColdPeers,
+                                     numberOfEstablishedPeers          = numberOfWarmPeers,
+                                     numberOfActivePeers               = numberOfHotPeers,
+
+                                     numberOfKnownBigLedgerPeers       = numberOfColdBigLedgerPeers,
+                                     numberOfEstablishedBigLedgerPeers = numberOfWarmBigLedgerPeers,
+                                     numberOfActiveBigLedgerPeers      = numberOfHotBigLedgerPeers,
+
+                                     numberOfKnownBootstrapPeers       = numberOfColdBootstrapPeers,
+                                     numberOfEstablishedBootstrapPeers = numberOfWarmBootstrapPeers,
+                                     numberOfActiveBootstrapPeers      = numberOfHotBootstrapPeers,
+
+                                     numberOfKnownLocalRootPeers       = numberOfColdLocalRootPeers,
+                                     numberOfEstablishedLocalRootPeers = numberOfWarmLocalRootPeers,
+                                     numberOfActiveLocalRootPeers      = numberOfHotLocalRootPeers,
+                                     localRoots                        = localRootsHWC,
+
+                                     numberOfKnownSharedPeers          = numberOfColdSharedPeers,
+                                     numberOfEstablishedSharedPeers    = numberOfWarmSharedPeers,
+                                     numberOfActiveSharedPeers         = numberOfHotSharedPeers })
```

Comment:
@karknu if you think we should trace `known` / `established` / `active` rather than `cold` / `warm` and `hot` I'll remove this pattern synonym.

Reply:
I'll keep it so we keep logging `cold` / `warm` / `hot`.  We'll remove it in a future version.  Otherwise all SPOs would need to adapt their grafana configuration in this version.